### DISCUSSION TITLE
Why I can't use "rebase" to update UEGitPlugin when as a submodule?

### DISCUSSION THREAD

**Discussion by ameaninglessname** [ProjectBorealis/UEGitPlugin#117] [Q&A]
error message when rebasing:
> From https://github.com/ProjectBorealis/UEGitPlugin
> 0e4c378..3382767  release    -> origin/release
> Auto-merging README.md
> CONFLICT (content): Merge conflict in README.md
> error: could not apply 1042e59... Documentation updates only.
> hint: Resolve all conflicts manually, mark them as resolved with
> hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
> hint: You can instead skip this commit: run "git rebase --skip".
> hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
> Could not apply 1042e59... Documentation updates only.
> fatal: Unable to rebase '40e1a595f0d31ddd26d35510cccf4d275d37c84b' in submodule path 'Plugins/UEGitPlugin'

I'm using the `TortoiseGit` GUI for git.
![image](https://github.com/ProjectBorealis/UEGitPlugin/assets/20263103/0ac2d849-9b78-4b2d-ada9-12ad4843b0ef)
![image](https://github.com/ProjectBorealis/UEGitPlugin/assets/20263103/f00c44fb-de89-40f0-83a5-63744e5779b9)

**Comment by mastercoms** [comment]
That's very strange, I am not sure. What does the conflict itself look like? And what was your `git reflog`?

**Reply by ameaninglessname** [reply]
![image](https://github.com/ProjectBorealis/UEGitPlugin/assets/20263103/237aa50c-38c5-40dd-841e-b9c78b055f4e)

**Reply by ameaninglessname** [reply]
the conflict:
![conflict](https://github.com/ProjectBorealis/UEGitPlugin/assets/20263103/9e7989d0-efc0-434d-8ab2-e14cad94c396)

new ref log after redo `git.exe submodule update --progress --rebase --remote --"Plugins/UEGitPlugin"`:
![new ref log](https://github.com/ProjectBorealis/UEGitPlugin/assets/20263103/4dbb70f6-004d-49d6-a4bf-2686a203b74f)

**Reply by mastercoms** [reply]
That's very strange, I am not sure why the history has diverged for you like that
